torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,189 @@ class Sentinel(RasterDataset):
     """
 
 
+class Sentinel1(Sentinel):
+    """Sentinel-1 dataset.
+
+    The `Sentinel-1 mission
+    <https://sentinel.esa.int/web/sentinel/missions/sentinel-1>`_ comprises a
+    constellation of two polar-orbiting satellites, operating day and night
+    performing C-band synthetic aperture radar imaging, enabling them to
+    acquire imagery regardless of the weather.
+
+    Data can be downloaded from:
+
+    * `Copernicus Open Access Hub
+      <https://scihub.copernicus.eu/>`_
+    * `Alaska Satellite Facility (ASF) Distributed Active Archive Center (DAAC)
+      <https://asf.alaska.edu/>`_
+    * `Microsoft's Planetary Computer
+      <https://planetarycomputer.microsoft.com/dataset/sentinel-1-rtc>`_
+
+    Product Types:
+
+    * `Level-0
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-0>`_:
+      Raw (RAW)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Single Look Complex (SLC)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Ground Range Detected (GRD)
+    * `Level-2
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-2>`_:
+      Ocean (OCN)
+
+    Polarizations:
+
+    * HH: horizontal transmit, horizontal receive
+    * HV: horizontal transmit, vertical receive
+    * VV: vertical transmit, vertical receive
+    * VH: vertical transmit, horizontal receive
+
+    Acquisition Modes:
+
+    * `Stripmap (SM)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/stripmap>`_
+    * `Interferometric Wide (IW) swath
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/interferometric-wide-swath>`_
+    * `Extra Wide (EW) swatch
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/extra-wide-swath>`_
+    * `Wave (WV)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/wave>`_
+
+    .. note::
+       At the moment, this dataset only supports the GRD product type. Data must be
+       radiometrically terrain corrected (RTC). This can be done manually using a DEM,
```

Comment:
Have you tested this with GRD and RTC data downloaded from the Planetary Computer? https://planetarycomputer.microsoft.com/dataset/group/sentinel-1

Reply:
Nope, please do! I can never remember the syntax to download files from PC. If the filename structure or file format is different, let me know and I can add more tests.

Reply:
You just go to the explorer, select either S1 dataset, click a scene, and click the code snippet:

![image](https://user-images.githubusercontent.com/1088078/208527175-277f7686-76a9-4d74-9d1b-22fb642778ad.png)

```
import pystac
import planetary_computer
import rioxarray

item_url = "https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-1-grd/items/S1A_IW_GRDH_1SDV_20221218T210639_20221218T210704_046391_058EB1"

# Load the individual item metadata and sign the assets
item = pystac.Item.from_file(item_url)
signed_item = planetary_computer.sign(item)

# Open one of the data assets (other asset keys to use: 'vv')
asset_href = signed_item.assets["vh"].href
```

From -- https://planetarycomputer.microsoft.com/explore?c=124.0274%2C-16.4940&z=9.00&v=2&d=sentinel-1-grd&m=Most+recent+-+VV%2C+VH&r=VV%2C+VH+False-color+composite&s=false%3A%3A100%3A%3Atrue&sr=desc&ae=0

Reply:
I might add this to our Wiki so I don't keep forgetting it...

Reply:
https://github.com/microsoft/torchgeo/wiki/Downloading-Data

Will try to update that with other sites I've used over the years

Reply:
Files on the Planetary Computer have the following URL:

> https://sentinel1euwestrtc.blob.core.windows.net/sentinel1-grd-rtc/GRD/2016/11/9/IW/DH/S1B_IW_GRDH_1SDH_20161109T042206_20161109T042231_002877_004E0A_7013/measurement/iw-hh.rtc.tiff

When downloaded, the file is named `iw-hh.rtc.tiff`. Our current RasterDataset implementation only supports parsing the filename, not the directory. And if you download the data locally, you don't have access to the rest of the URL anyway. So this data loader likely works with Planetary Computer data, but requires you to manually rename all files to match the format used by ASF DAAC.

Reply:
Also, this dataset likely doesn't work for the GRD files, they have to first be RTC'ed to convert the ground control points to a CRS.

Reply:
Okay, I confirmed that it works with Planetary Computer RTC data as long as you rename the files. Also confirmed that both the HH/HV and HH plots work correctly.